File: src/BetterLyrics.Core/Interfaces/Services/ISettingsService.cs
// 2025/6/23 by Zhe Fang

using BetterLyrics.Core.Models.Settings;

namespace BetterLyrics.Core.Interfaces.Services;

public interface ISettingsService
{
    AppSettings AppSettings { get; set; }

    bool ImportSettings(string importPath);
    void ExportSettings(string exportPath);
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/interfaces/services/ISettingsService.kt
package com.jayfunc.betterlyrics.interfaces.services
import com.jayfunc.betterlyrics.models.settings.AppSettings
import com.jayfunc.betterlyrics.models.settings.LyricsSaveConfig
interface ISettingsService {
    var appSettings: AppSettings?
    var lyricsSaveConfig: LyricsSaveConfig?
    fun loadSettings()
    fun saveSettings()
    fun exportSettings(path: String)
    fun importSettings(path: String): Boolean
    fun resetSettings()
}
